queue_job/models/database_load_indicator.py

```diff
@@ -0,0 +1,171 @@
+# Copyright (C) 2018 DynApps <http://www.dynapps.be>
+# @author Stefan Rijnhart <stefan.rijnhart@dynapps.nl>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import logging
+from ast import literal_eval
+from time import sleep
+
+from psycopg2 import ProgrammingError
+from psycopg2.sql import SQL, Identifier
+
+from odoo import _, api, fields, models
+from odoo.exceptions import AccessError, UserError
+from odoo.tools import ormcache
+
+_logger = logging.getLogger(__name__)
+
+
+class DatabaseLoadIndicator(models.Model):
+    _name = "database.load.indicator"
+    _description = "Database Load Indicator"
+    _order = "sequence, name"
+
+    @api.model
+    def _default_sequence(self):
+        max_seq = self.search([], limit=1, order="sequence desc")
+        return (max_seq.sequence if max_seq else 0) + 1
+
+    name = fields.Char(required=True)
+    active = fields.Boolean(default=True)
+    debug = fields.Boolean(
+        default=False, help="Log the result of every call to this indicator"
+    )
+    sequence = fields.Integer(default=_default_sequence)
+    function = fields.Char(required=True)
+    sleep_before = fields.Integer(
+        default=1,
+        help=(
+            "How long to sleep before every call (i.e. after queuing the last "
+            "the last set of jobs."
+        ),
+    )
+    sleep = fields.Integer(
+        string="Sleep After",
+        default=3,
+        help=(
+            "How long to sleep when database is overloaded (in addition to "
+            '"Sleep Before"'
+        ),
+    )
+    factor = fields.Float(
+        default=-25,
+        help=(
+            "Multiplier before applying the resulting load indicator. If "
+            "the result is a negative number, it is returned as 0."
+        ),
+    )
+    arguments = fields.Char(default="()")
+    definition = fields.Text(
+        compute="_compute_definition",
+        help=(
+            "This is the definition as retrieved from the Odoo database. "
+            "You can only update the definition in the database itself."
+        ),
+    )
+
+    def _compute_definition(self):
```

Comment:
Other methods have a docstring, `_compute_definition` and `_get_active_indicator` don't.

Reply:
Thanks, docstring added.